frontend/app/src/lib/atoms.ts

```diff
@@ -88,16 +100,38 @@ export function useTenant(): TenantContext {
 
     // Finally, if neither a current tenant is set as a query param
     // nor if a tenant was set in Jotai, use the first membership as a fallback
-    const firstMembershipTenant = memberships?.[0]?.tenant;
+    const firstMembershipTenant = memberships.at(0)?.tenant;
 
     return firstMembershipTenant;
   }, [memberships, searchParams, findTenant, lastTenant]);
 
   const currTenantId = searchParams.get('tenant');
   const currTenant = currTenantId ? findTenant(currTenantId) : undefined;
 
+  const tenant = currTenant || computedCurrTenant;
+
+  // If the tenant is not set as a query param at any point,
+  // set it.
+  // NOTE: This is helpful mostly for debugging to easily grab
+  // the tenant from the URL.
+  useEffect(() => {
+    const currentTenantParam = searchParams.get('tenant');
+    if (!currentTenantParam && tenant) {
+      setTenant(tenant);
+    }
+  }, [searchParams, tenant, setTenant]);
+
+  if (!tenant) {
+    return {
+      tenant: undefined,
+      tenantId: undefined,
+      setTenant,
+    };
+  }
+
   return {
-    tenant: currTenant || computedCurrTenant,
+    tenant,
+    tenantId: tenant.metadata.id,
```

Comment:
this is just a helper to shorten needing to use `tenant.metadata.id` every time we want to grab the tenant